src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs

```diff
@@ -134,50 +126,51 @@ private static unsafe SafeMsQuicConfigurationHandle Create(QuicOptions options,
 
             Debug.Assert(!MsQuicApi.Api.Registration.IsInvalid);
 
-            var settings = new QuicSettings
-            {
-                IsSetFlags = QuicSettingsIsSetFlags.PeerBidiStreamCount |
-                             QuicSettingsIsSetFlags.PeerUnidiStreamCount,
-                PeerBidiStreamCount = (ushort)options.MaxBidirectionalStreams,
-                PeerUnidiStreamCount = (ushort)options.MaxUnidirectionalStreams
-            };
+            QUIC_SETTINGS settings = default(QUIC_SETTINGS);
+            settings.IsSet.PeerUnidiStreamCount = 1;
+            settings.PeerUnidiStreamCount = (ushort)options.MaxUnidirectionalStreams;
+            settings.IsSet.PeerBidiStreamCount = 1;
+            settings.PeerBidiStreamCount = (ushort)options.MaxBidirectionalStreams;
 
+            settings.IsSet.IdleTimeoutMs = 1;
             if (options.IdleTimeout != Timeout.InfiniteTimeSpan)
             {
                 if (options.IdleTimeout <= TimeSpan.Zero) throw new Exception("IdleTimeout must not be negative.");
-
-                ulong ms = (ulong)options.IdleTimeout.Ticks / TimeSpan.TicksPerMillisecond;
-                if (ms > (1ul << 62) - 1) throw new Exception("IdleTimeout is too large (max 2^62-1 milliseconds)");
-
                 settings.IdleTimeoutMs = (ulong)options.IdleTimeout.TotalMilliseconds;
             }
             else
             {
                 settings.IdleTimeoutMs = 0;
             }
-            settings.IsSetFlags |= QuicSettingsIsSetFlags.IdleTimeoutMs;
 
-            uint status;
-            SafeMsQuicConfigurationHandle? configurationHandle;
+            SafeMsQuicConfigurationHandle configurationHandle;
             X509Certificate2[]? intermediates = null;
 
             MemoryHandle[]? handles = null;
-            QuicBuffer[]? buffers = null;
+            // TODO: should they be pinned? The content is pinned, but the array holding the individual buffers isn't.
```

Comment:
Are you asking if the buffers need to stay alive longer than the call to `ConfigurationOpen`? The answer to that question is no.

Reply:
It's not about lifetime, but rather about not being moved by GC. Pinning makes sure that GC will not move it when it does compaction. We pin the individual buffers within the `QUIC_BUFFER[]` but we don't pin this holding array itself. Which I think is wrong and is only working by pure chance, but I'm not 100% sure. @jkotas maybe you can shed a light here?

Reply:
Ah, ok. Thanks for explaining. Yeah, moving any part of the memory you pass to MsQuic would be bad.

Reply:
Yes, this is problem. It may be best to allocate `buffers` as unmanaged memory block.